main.go

```diff
@@ -0,0 +1,181 @@
+	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stdout})
+	app := cli.NewApp()
+	app.Name = "web3-cli"
+	app.Version = "0.0.1"
+	app.Usage = "web3 cli tool"
+	globalFlags := []cli.Flag{
+		cli.StringFlag{
+			Name:        "network",
+			Usage:       "The name of the network (gochain-testnet/gochain-mainnet/ethereum/localhost)",
+			Value:       "gochain-testnet",
+			Destination: &network,
+			EnvVar:      "NETWORK",
+			Hidden:      false},
+		cli.StringFlag{
+			Name:        "rpc-url",
+			Usage:       "The network RPC URL",
+			Destination: &rpcUrl,
+			EnvVar:      "RPC_URL",
+			Hidden:      false},
+	}
+	app.Commands = []cli.Command{
+		{
+			Name:    "block",
+			Usage:   "Show information about the block",
+			Aliases: []string{"bl"},
+			Flags:   globalFlags,
+			Action: func(c *cli.Context) {
+				GetBlockDetails(network, rpcUrl, c.Args().First())
+			},
+		},
+		{
+			Name:    "transaction",
+			Aliases: []string{"tx"},
+			Flags:   globalFlags,
+			Usage:   "Show information about the transaction",
+			Action: func(c *cli.Context) {
+				GetTransactionDetails(network, rpcUrl, c.Args().First())
+			},
+		},
+		{
+			Name:    "address",
+			Aliases: []string{"addr"},
+			Flags:   globalFlags,
+			Usage:   "Show information about the address",
+			Action: func(c *cli.Context) {
+				GetAddressDetails(network, rpcUrl, c.Args().First())
+			},
+		},
+		{
+			Name:    "contract",
+			Aliases: []string{"c"},
+			Flags:   globalFlags,
+			Usage:   "actions with contracts",
+			Subcommands: []cli.Command{
+				{
+					Name:  "build",
+					Usage: "Build the specified contract",
+					Action: func(c *cli.Context) {
+						BuildSol(c.Args().First())
+					},
+				},
+				{
+					Name:  "deploy",
+					Usage: "Build and deploy the specified contract to the network",
+					Action: func(c *cli.Context) {
+						fmt.Println("deploying the contract from the file: ", c.Args().First())
+					},
+				},
+				{
+					Name:  "call",
+					Usage: "Call the specified function of the contract",
+					Action: func(c *cli.Context) {
+						println("calling the function of the deployed contract")
+						fmt.Println("calling the function of the deployed contract from:", network)
+					},
+					Flags: []cli.Flag{
+						cli.StringFlag{
+							Name:        "function",
+							Usage:       "The name of the function to call",
+							Destination: &function,
+							Hidden:      false},
+						cli.StringFlag{
+							Name:        "contract",
+							Destination: &contractAddress,
+							Usage:       "The address of the deployed contract",
+							Hidden:      false},
+					},
+				},
+			},
+		},
+	}
+	app.Run(os.Args)
+}
+
+func getRPCURL(network, rpcURL string) string {
+
+	if rpcURL != "" {
+		return rpcURL
+	}
+
+	switch network {
+	case "gochain-mainnet":
+		return "https://testnet-rpc.gochain.io"
+	case "localhost":
+		return "localhost"
+	case "ethereum":
+		return "https://main-rpc.linkpool.io"
+	default:
+		//gochain-mainnet
```

Comment:
It might be better to error out on a mismatch, and only match `mainnet` or `gochain-mainnet` here, then they won't e.g. accidentally spend money deploying a contract when they typo `testnte`.